python/ray/ml/trainer.py

```diff
@@ -31,13 +32,99 @@ class TrainingFailedError(RuntimeError):
 class Trainer(abc.ABC):
```

Comment:
Btw, all of Trainer should be DeveloperAPI, no? It's intended for library authors. End users will use TF/Torch trainers.

Reply:
Yeah good point- updated!